tests/cancun/eip6780_selfdestruct/test_dynamic_create2_selfdestruct_collision.py

```diff
@@ -226,6 +228,247 @@ def test_dynamic_create2_selfdestruct_collision(
+            + Op.SSTORE(
+                first_create2_result,
+                Op.MLOAD(0),
+            )
+            # In case the create2 didn't work, flush account balance
+            + Op.CALL(100000, address_code, 0, 0, 0, 0, 0)
+            # Call to the created account to trigger selfdestruct
+            + Op.CALL(100000, create2_address, first_call_value, 0, 0, 0, 0)
+            + Op.SSTORE(code_worked, 1),
+            storage={first_create2_result: 0xFF},
+        ),
+        address_to_second: Account(
+            balance=100000000,
+            nonce=0,
+            code=Op.JUMPDEST()
+            # Make a subcall that do CREATE2 collision and returns its address as the result
+            + Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
+            + Op.CALL(100000, address_code, second_create2_value, 0, Op.CALLDATASIZE(), 0, 32)
+            + Op.SSTORE(
+                second_create2_result,
+                Op.MLOAD(0),
+            )
+            # Call to the created account to trigger selfdestruct
+            + Op.CALL(200000, call_address_in_the_end, second_call_value, 0, 0, 0, 0)
+            + Op.SSTORE(code_worked, 1),
+            storage={second_create2_result: 0xFF},
+        ),
+        address_code: Account(
+            balance=0,
+            nonce=0,
+            code=Op.CALLDATACOPY(0, 0, Op.CALLDATASIZE())
+            + Op.MSTORE(
+                0,
+                Op.CREATE2(Op.SELFBALANCE(), 0, Op.CALLDATASIZE(), create2_salt),
+            )
+            + Op.RETURN(0, 32),
+            storage={},
+        ),
+        address_create2_storage: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code=Op.SSTORE(1, 1),
+            storage={},
+        ),
+        TestAddress: Account(
+            balance=7000000000000000000,
+            nonce=0,
+            code="0x",
+            storage={},
+        ),
+    }
+
+    if create2_dest_already_in_state:
+        # Create2 address already in the state, e.g. deployed in a previous block
+        pre[create2_address] = Account(
+            balance=pre_existing_create2_balance,
+            nonce=1,
+            code=deploy_code,
+            storage={},
+        )
+
+    post: Dict[Address, Union[Account, object]] = {}
+
+    # Create2 address only exists if it was pre-existing and after cancun
+    post[create2_address] = (
+        Account(balance=0, nonce=1, code=deploy_code, storage={create2_constructor_worked: 0x00})
+        if create2_dest_already_in_state and fork >= Cancun
+        else Account.NONEXISTENT
+    )
+
+    # after Cancun Create2 initcode is only executed if the contract did not already exist
+    # and before it will always be executed as the first tx deletes the account
+    post[address_create2_storage] = Account(
+        storage={
+            create2_constructor_worked: (
+                1 if fork < Cancun else int(not create2_dest_already_in_state)
```

Comment:
```suggestion
                int(fork < Cancun or not create2_dest_already_in_state)
```
I'd reword it to something like this but it's just preference.